src/openms/include/OpenMS/ANALYSIS/ID/FalseDiscoveryRate.h

```diff
@@ -73,27 +75,67 @@ namespace OpenMS
     void apply(std::vector<PeptideIdentification>& fwd_ids, std::vector<PeptideIdentification>& rev_ids) const;
 
     /**
-        @brief Calculates the FDR of one run from a concatenated sequence DB search
+    @brief Calculates the FDR of one run from a concatenated sequence DB search
 
-        @param id peptide identifications, containing target and decoy hits
+    @param id peptide identifications, containing target and decoy hits
     */
     void apply(std::vector<PeptideIdentification>& id) const;
 
     /**
-       @brief Calculates the FDR of two runs, a forward run and decoy run on protein level
+    @brief Calculates the FDR of two runs, a forward run and decoy run on protein level
 
-       @param fwd_ids forward protein identifications
-       @param rev_ids reverse protein identifications
+    @param fwd_ids forward protein identifications
+    @param rev_ids reverse protein identifications
     */
     void apply(std::vector<ProteinIdentification>& fwd_ids, std::vector<ProteinIdentification>& rev_ids) const;
 
     /**
-       @brief Calculate the FDR of one run from a concatenated sequence db search
+    @brief Calculate the FDR of one run from a concatenated sequence db search
 
-       @param ids protein identifications, containing target and decoy hits
+    @param ids protein identifications, containing target and decoy hits
     */
     void apply(std::vector<ProteinIdentification>& ids) const;
 
+    /**
+    @brief Calculate the FDR based on PEPs or PPs (if present) and modifies the IDs inplace
+
+    @param ids protein identifications, containing PEP scores (not necessarily) annotated with target decoy.
+    */
+    void applyEstimated(std::vector<ProteinIdentification>& ids) const;
+
+    /**
+    @brief Calculate a linear combination of the area of the difference in estimated vs. empirical (TD) FDR
+     and the ROC-N value (AUC up to first N false positives).
+
+    @param ids protein identifications, containing PEP scores annotated with target decoy. If vector, only first will be evaluated-
+    @param pepCutoff up to which PEP should the differences between the two FDRs be calculated
+    @param fpCutoff up to which nr. of false positives should the target-decoy AUC be evaluated
+    @param diffWeight which weight should the difference get. The ROC-N value gets 1 - this weight.
+    */
+    double applyEvaluateProteinIDs(const std::vector<ProteinIdentification>& ids, double pepCutoff = 1.0, UInt fpCutoff = 50, double diffWeight = 0.2);
+    double applyEvaluateProteinIDs(const ProteinIdentification& ids, double pepCutoff = 1.0, UInt fpCutoff = 50, double diffWeight = 0.2);
+
+    /// simpler reimplemetation of the apply function above.
```

Comment:
more documentation.
difference to apply?

Reply:
There should be at least one place with the e.g.  t/(d+1) formula

Reply:
I never understood the old one, so I can't write about differences ;)
Added formula and more description.